google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Fields.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery;
+
+import com.google.api.services.bigquery.model.TableFieldSchema;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import java.io.Serializable;
+import java.util.AbstractList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Google BigQuery Table schema fields (columns). Each field has a unique name and index. Fields
+ * with duplicate names are not allowed in BigQuery schema.
+ */
+public final class Fields extends AbstractList<Field> implements Serializable {
```

Comment:
It seems like this should be called `FieldList`

Reply:
I like FieldValues and Fields more, as they are shorter and still clear. But I changed to FieldList and FieldValueList to make clear that those are valid lists.